src/test/java/de/tum/in/www1/artemis/programmingexercise/CourseBitbucketBambooJiraIntegrationTest.java

```diff
@@ -207,16 +209,18 @@ void testGetCoursesWithQuizExercises() throws Exception {
         courseTestService.testGetCoursesWithQuizExercises();
     }
 
-    @Test
+    @ParameterizedTest(name = "{displayName} [{index}] {argumentsWithNames}")
     @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
-    void testGetCourseForDashboard() throws Exception {
-        courseTestService.testGetCourseForDashboard(false);
+    @ValueSource(booleans = { true, false })
+    void testGetCourseForDashboard(boolean userRefresh) throws Exception {
+        courseTestService.testGetCourseForDashboard(userRefresh);
     }
 
-    @Test
-    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
-    void testGetCourseForDashboard_userRefresh() throws Exception {
-        courseTestService.testGetCourseForDashboard(true);
+    @ParameterizedTest(name = "{displayName} [{index}] {argumentsWithNames}")
+    @WithMockUser(username = TEST_PREFIX + "custom1", roles = { "USER", "TA", "EDITOR", "INSTRUCTOR" })
```

Comment:
I haven't seen this syntax so far. Does this test with all different user roles once? Otherwise, it should be enough testing with the highest role, shouldn't it?

Reply:
Same in the next test file

Reply:
The user has all these roles, since he is part of multiple courses, each with a different role.
So in my understanding he should have all of them